types/chrome/index.d.ts

```diff
@@ -12562,31 +12562,122 @@ declare namespace chrome.declarativeNetRequest {
+            tabId?: number;
+            /**
+             * The window in which to open the side panel.
+             * This is only applicable if the extension has a global (non-tab-specific) side panel or tabId is also specified.
+             * This will override any currently-active global side panel the user has open in the given window.
+             * At least one of this and tabId must be provided.
+             */
+            windowId?: number;
+        }
+        & ({
+            tabId: number;
+        } | {
+            windowId: number;
+        });
+
     export interface PanelBehavior {
+        /** Whether clicking the extension's icon will toggle showing the extension's entry in the side panel. Defaults to false. */
         openPanelOnActionClick?: boolean;
     }
 
     export interface PanelOptions {
+        /** Whether the side panel should be enabled. This is optional. The default value is true. */
         enabled?: boolean;
+        /** The path to the side panel HTML file to use. This must be a local resource within the extension package. */
         path?: string;
+        /**
+         * If specified, the side panel options will only apply to the tab with this id.
+         * If omitted, these options set the default behavior (used for any tab that doesn't have specific settings).
+         * Note: if the same path is set for this tabId and the default tabId, then the panel for this tabId will be a different instance than the panel for the default tabId.
+         */
         tabId?: number;
     }
 
     export interface SidePanel {
+        /** Developer specified path for side panel display. */
         default_path: string;
     }
 
+    /**
+     * Returns the active panel configuration.
+     * Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility.
+     * You cannot use both on the same function call.
+     * The promise resolves with the same type that is passed to the callback.
+     */
     export function getOptions(
+        /** Specifies the context to return the configuration for. */
         options: GetPanelOptions,
+        /** The `callback` parameter looks like: ```(options: PanelOptions) => void``` */
         callback?: (options: PanelOptions) => void,
     ): Promise<PanelOptions>;
 
-    export function getPanelBehavior(callback?: (behavior: PanelBehavior) => void): Promise<PanelBehavior>;
+    /**
+     * Returns the extension's current side panel behavior.
+     * Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility.
+     * You cannot use both on the same function call.
+     * The promise resolves with the same type that is passed to the callback.
+     */
+    export function getPanelBehavior(
+        /** The `callback` parameter looks like: `(behavior: PanelBehavior) => void` */
+        callback?: (behavior: PanelBehavior) => void,
+    ): Promise<PanelBehavior>;
```

Comment:
I believe this should show that the callback can't be used when you want a promise. Ideally could you do this for each of the functions you've modified?

IMO you can omit the parameter comments if they only describe the type - Editors should see the types already. I'm fine with leaving them though. 

Note: I haven't tried to build and test these suggestions.

```suggestion
    export function getPanelBehavior(
        callback: (behavior: PanelBehavior) => void) => void;
        export function getPanelBehavior(
    ): Promise<PanelBehavior>;
```

Reply:
I agree the parameter comments for the callbacks are a bit redundant, I only added it because that's the note that is made in the official chrome documentation. I don't have any strong opinions about leaving them in.